source/pkcs11.c

```diff
@@ -1177,6 +1187,136 @@ static int s_pkcs11_sign_rsa(
     return success ? AWS_OP_SUCCESS : AWS_OP_ERR;
 }
 
+/*
+ * Basic ASN.1 (DER) encoding of header -- sufficient for ECDSA
+ */
+static int s_asn1_enc_prefix(struct aws_byte_buf *buffer, uint8_t identifier, size_t length) {
+    if (((identifier & 0x1f) == 0x1f) || (length > 0x7f)) {
+        AWS_LOGF_ERROR(AWS_LS_IO_PKCS11, "Unable to encode ASN.1 (DER) header 0x%02x %lu", identifier, (long)length);
```

Comment:
I was wondering about that - I didn't see %zu used elsewhere (probably didn't look hard enough), and wasn't sure if it would break in older versions of C compiler.